src/main/java/com/alkemy/ong/config/AWSConfiguration.java

```diff
@@ -0,0 +1,47 @@
+package com.alkemy.ong.config;
+
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.regions.RegionUtils;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class AWSConfiguration {
+
+    @Value("$ {aws_endpoint_URL}")
+    private String endpointURL;
+
+    @Value("$ {aws_bucket}")
+    private String bucketName;
+
+    @Value("$ {aws_access_key}")
+    private String accessKey;
+
+    @Value("$ {aws_secret_key}")
+    private String secretKey;
+
+    @Value("$ {aws_region}")
+    private String region;
+
+
+    public String getBucketName() {
+        return this.bucketName;
+    }
+
+    @Bean
+    public AmazonS3 initializeAmazon() {
+        BasicAWSCredentials credentials = new BasicAWSCredentials(this.accessKey, this.secretKey);
+        
+        return AmazonS3ClientBuilder.standard().withRegion(Regions.DEFAULT_REGION)
```

Comment:
Tengo dudas sobre como manejar .withRegion. Si bien en varias fuentes encontré que .withRegion acepta tanto  enumerado como un string al hacer, lo presentan de esta manera:
.withRegion(Regions.fromName(region))
.withRegion(region)
usando cualquiera de ellos tira el error “nested exception is com.amazonaws.SdkClientException: Unable to find a region via the region provider chain. Must provide an explicit region in the builder”

Las dos formas de resolver fueron:
Harcodear: .
withRegion(Regions. Regions.US_EAST_1)
withRegion(Regions.DEFAULT_REGION)
Esta ultima el default región es US_EAST_1 y esta explicado en la seccion “Automatically Determine the Region from the Environment” del siguiente link
https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/java-dg-region-selection.html
acá dejo otro links que use por si acaso.
https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/setup-credentials.html
https://stackoverflow.com/a/51362437

**Por ultimo, de quedar así entonces debería sacar el atributo región. Espero tu corrección al respecto** 😊

Reply:
.withRegion(Regions.fromName(region)) esta más que perfecto, pero no te esta funcionando porque lo pusiste en mayúsculas en el .yaml y si revisas la implementación de Regions#fromName, los nombres del enum estan en minúsculas y esta utilizando el operador equals, por lo que si lo escribís en mayúscula ... BOOM. Por convención, todo lo que es setup de cualquier cosa va en minúsculas.